adagrams/game.py

```diff
@@ -1,11 +1,151 @@
+            letter_pool_copy[random_letter] -= 1         
+    return draw
 
 def uses_available_letters(word, letter_bank):
-    pass
+    #check if the word is an anagram of some or all of the given letters in the hand
+
+    letter_bank_copy = letter_bank.copy()
+    word = word.upper()  
+    for char in word:
+        #Take one of the letters from the hand and make unavailable in the bank copy
+        if char in letter_bank_copy:
+            letter_bank_copy.remove(char)
+        else:
+            return False
 
+    return True
+
+#-------- Score-Word Func Using Nested Lists --------#
+SCORE_CHART = [
+    [1,['A', 'E','I','O','U','L','N','R','S','T']],
+    [2,['D','G']],
+    [3,['B','C','M','P']],
+    [4,['F','H','V','W','Y']],
+    [5,['K']],
+    [8,['J','X']],
+    [10,['Q','Z']]
+]
 def score_word(word):
-    pass
+    #Returns the score of a given word as defined by the SCORE_CHART
+
+    total_points = 0
+    points = 0
+    word = word.upper()
+    for i in range(len(SCORE_CHART)):
+        for char in word:
+            if char in SCORE_CHART[i][1]: #check for char in each letter in list rows
+                points =SCORE_CHART[i][0] #If found, access the index that has the score
+                total_points += points
+            else:
+                points = 0
+
+    #Extra points if word has 7 or more characters 
+    if len(word) >= 7:
+        total_points += 8
+    
+    return total_points
+
+# #------Score_Word Func Using Dictionaries --------#
+# SCORE_CHART = {
+#     1:['A', 'E','I','O','U','L','N','R','S','T'],
+#     2:['D','G'],
+#     3:['B','C','M','P'],
+#     4:['F','H','V','W','Y'],
+#     5:['K'],
+#     8:['J','X'],
+#     10:['Q','Z']
+
+# }
+# def score_word(word):
+#     #Returns the score of a given word as defined by the SCORE_CHART
+
+#     points = 0
+#     word = word.upper()
+
+#     #access the lists in the dict (values)as letters
+#     for score, letters in SCORE_CHART.items():
+#         for i in range(len(letters)): 
+#             for char in word:        
+#                 if char in letters[i]: #check char in
+#                     points += score    
+
+#     #Extra points if word has 7 or more characters 
+#     if len(word) >= 7:
+#         points += 8
+
+#     return points
 
 def get_highest_word_score(word_list):
-    pass
\ No newline at end of file
+    #Calculate which word from the list has the highest score 
+    # applying tie-breaking logic
+    
+    highest_score = 0
+    highest_word = " "
+    
+    #calculate score for word in word_list
+    for word in word_list:
+        current_score = score_word(word) 
+        
+        #choose the word with highest score
+        if current_score > highest_score:
+            highest_word= word
+            highest_score = current_score
+        
+        #If a tie and no word is 10 char, pick the shortest word 
+        if current_score == highest_score \
```

Comment:
I would change this to an `elif` so that this line of code and the other `elif` statement below don't run when line `136` is not `True`. Right now both of these conditions are still checked when though they will never run if `current_score > highest_score`. It's inconsequential here, but could cause some logical errors in other cases.

Reply:
That makes a lot of sense!! Thank you so much for your feedback, very much appreciated 🤝